Index: financial-system/kfs/branches/mod-hlo/work/src/edu/arizona/kfs/pdp/batch/LoadAchPayeeBankAcctStep.java
===================================================================
--- financial-system/kfs/branches/mod-hlo/work/src/edu/arizona/kfs/pdp/batch/LoadAchPayeeBankAcctStep.java	(revision 9944)
+++ financial-system/kfs/branches/mod-hlo/work/src/edu/arizona/kfs/pdp/batch/LoadAchPayeeBankAcctStep.java	(revision 9945)
@@ -21,7 +21,7 @@
 import java.util.List;
 
 import org.apache.commons.lang.StringUtils;
-import org.kuali.kfs.pdp.service.PayeeAchAccountService;
+import edu.arizona.kfs.pdp.service.PayeeAchAccountService;
 import org.kuali.kfs.sys.batch.AbstractStep;
 import org.kuali.kfs.sys.batch.BatchInputFileType;
 import org.kuali.kfs.sys.batch.service.BatchInputFileService;
@@ -44,14 +44,14 @@
 
         boolean processSuccess = true;
         List<String> processedFiles = new ArrayList();
-/**        for (String inputFileName : fileNamesToLoad) {
-            processSuccess = payeeAchAccountService.loadAchBankFile(inputFileName);
+        for (String inputFileName : fileNamesToLoad) {
+            processSuccess = payeeAchAccountService.loadAchPayeeAccountFile(inputFileName);
             if (processSuccess) {
                 processedFiles.add(inputFileName);
             }
-        } **/
+        }
 
-//        removeDoneFiles(processedFiles);
+        removeDoneFiles(processedFiles);
 
         return processSuccess;
     }
